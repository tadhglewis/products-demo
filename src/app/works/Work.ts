type WorkUrl = {
  type: 'small' | 'medium' | 'large';
  link: string;
};

export type Exif = {
  model: string;
  software: string;
  isoSpeedRatings: string;
  dateTime: string;
  make: string;
};

type Work = {
  id: string;
  filename: string;
  imageWidth: string;
  imageHeight: string;
  urls: WorkUrl[];
  exif: Exif;
};

export default Work;
