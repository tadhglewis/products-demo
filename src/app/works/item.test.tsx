import { fireEvent, render, screen } from '@testing-library/react';
import Item from './item';
import Work from './Work';

describe('item', () => {
  const work: Work = {
    id: '31820',
    filename: '162042.jpg',
    imageWidth: '2206',
    imageHeight: '2205',
    urls: [
      {
        type: 'small',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,135x135,075,f.jpg',
      },
      {
        type: 'medium',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,300x300,075,f.jpg',
      },
      {
        type: 'large',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,550x550,075,f.jpg',
      },
    ],
    exif: {
      model: 'NIKON D80',
    },
  };

  it('should match snapshot', async () => {
    const { container } = render(<Item {...work} addToFilter={jest.fn()} />);

    fireEvent.click(screen.getByText('Model: NIKON D80'));

    expect(container).toMatchSnapshot();
  });

  it('should call addToFilter when detail is clicked', () => {
    const addToFilterMock = jest.fn();

    render(<Item {...work} addToFilter={addToFilterMock} />);

    fireEvent.click(screen.getByText('Model: NIKON D80'));

    expect(addToFilterMock).toHaveBeenCalledWith('model', 'NIKON D80');
  });
});
