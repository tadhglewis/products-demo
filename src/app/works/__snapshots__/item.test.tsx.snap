// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item should match snapshot 1`] = `
.c0 {
  border: 1px solid rgba(0,0,0,0.1);
  border-radius: 3px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 150px;
  height: 150px;
  overflow: hidden;
  position: relative;
}

.c2 {
  display: block;
  height: 100%;
  position: absolute;
}

.c3 {
  padding: 16px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  border: 1px solid rgba(0,0,0,0.1);
  padding: 0 8px;
  border-radius: 12px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c4:hover {
  background-color: rgb(69 134 255 / 10%);
}

.c5 {
  padding: 16px;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="http://ih1.redbubble.net/work.31820.1.flat,550x550,075,f.jpg"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="162042.jpg"
        class="c2"
        src="http://ih1.redbubble.net/work.31820.1.flat,135x135,075,f.jpg"
      />
    </a>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        Model
        : 
        NIKON D80
      </div>
    </div>
    <div
      class="c5"
    >
      30/05/2021
    </div>
  </div>
</div>
`;
